spinnaker/spinnaker.go

```diff
@@ -479,12 +479,35 @@ func (s Spinnaker) asgs(appName, account, clusterName string) (result []spinnake
 	return asgs, nil
 }
 
-// CloudProvider returns the cloud provider for a given account
-func (s Spinnaker) CloudProvider(account string) (provider string, err error) {
-	url := s.accountURL(account)
+// CloudProvider returns the cloud provider for a given account name
+func (s Spinnaker) CloudProvider(name string) (provider string, err error) {
+	account, err := s.account(name)
+	if err != nil {
+		return "", err
+	}
+
+	if account.CloudProvider == "" {
+		return "", fmt.Errorf("no cloudProvider field in response body")
```

Comment:
If you don't need to format your return error, `errors.new("")` will be fine :)

Reply:
Thanks !